src/Chainweb/Miner/Coordinator.hs

```diff
@@ -224,8 +227,14 @@ chainChoice c choice = case choice of
   where
     loop :: ChainId -> IO ChainId
     loop cid = do
-        new <- randomChainIdAt c (minChainHeight c)
-        bool (pure new) (loop cid) $ new == cid
+        if _chainGraph c == singletonChainGraph
+        then do
+            let millisecondInMicros = 1_000
+            threadDelay millisecondInMicros
```

Comment:
What about using another `ChainChoice` in case of `singletonChainGraph`? Or, alternatively, waiting for a new cut to appear?

Reply:
How about waiting until the PrimeWork TVar changes?